src/core/core.scale.js

```diff
@@ -693,8 +708,13 @@ module.exports = function(Chart) {
 						labelY = me.bottom - labelYOffset;
 					}
 
-					var xLineValue = me.getPixelForTick(index) + helpers.aliasPixel(lineWidth); // xvalues for grid lines
-					labelX = me.getPixelForTick(index, gridLines.offsetGridLines) + optionTicks.labelOffset; // x values for optionTicks (need to consider offsetLabel option)
+					var xLineValue = getLineValue(me, index, gridLines.offsetGridLines && ticks.length > 1);
+					if (xLineValue < me.left) {
+						lineColor = 'rgba(0,0,0,0)';
```

Comment:
This prevents the leftmost grid line from being drawn outsize the chart area like this.

<img width="387" alt="screen shot 2017-08-12 at 3 35 45 am" src="https://user-images.githubusercontent.com/723188/29229146-6c97b678-7f0f-11e7-973d-fea7b3dd5baf.png">